src/components/m-form-group/_template.js

```diff
@@ -11,20 +11,20 @@ export default function ({
 
   return [
     html`
-      <label class="m-form-group__wrapper js-form-group__wrapper"> 
-        <span class="m-form-group__label-icon-wrapper js-form-group__label-icon-wrapper">
-          <span class="m-form-group__label">
+      <fieldset class="m-form-group__wrapper js-form-group__wrapper">
```

Comment:
If the `<legend>` below has a class element suffix of `__legend`, I would do the same for the `<fieldset>` replacing `wrapper` by `filedset` 🤔